lib/RudderProperty.dart

```diff
@@ -14,15 +14,15 @@ class RudderProperty {
   }
 
   RudderProperty putValue(
-      {String key, dynamic value, Map<String, dynamic> map}) {
+      {String? key, dynamic value, Map<String?, dynamic>? map}) {
     if (map != null) {
       this.__map.addAll(map);
       return this;
     }
     if (value is RudderProperty) {
       this.__map[key] = value.getMap();
     } else {
-      map[key] = value;
+      map![key] = value;
     }
     return this;
```

Comment:
Can we please convert it into this way @abed-binary : 
```suggestion
  {String? key, dynamic value, Map<String, dynamic>? map}) {
    if (map != null) {
      this.__map.addAll(map);
      return this;
    }
    if (key != null) {
      if (value is RudderProperty) {
        this.__map[key] = value.getMap();
      } else {
        this.__map[key] = value;
      }
    }
    return this;
  }
```